warp10/src/main/java/io/warp10/continuum/egress/EgressExecHandler.java

```diff
@@ -321,12 +399,22 @@ public void handle(String target, Request baseRequest, HttpServletRequest req, H
         times.add(end - nano);
       }
 
+      if (forcedMacro) {
+        stack.macroClose();
+      }
+
       //
       // Make sure stack is balanced
       //
 
       stack.checkBalanced();
 
+      if (maxtime > 0) {
+        stack.push(maxtime);
+        TIMEBOX.apply(stack);
+        stack.handleSignal();
```

Comment:
I don't see why you need to handleSignal here. Could you add a comment to explain?